# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1207339051 14400
#      Fri Apr 04 15:57:31 2008 -0400
# Branch release-3-0-x
# Node ID 5d011c1f68fd6e26c06439cb9d37cd1dccbd0159
# Parent  5dc5de967ab08ae3123e36faee9f7351d409f89a
make_constant: handle escape sequences in dq-strings

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2008-04-04  John W. Eaton  <jwe@octave.org>
+
+	* parse.y (make_constant): Handle escape sequences in dq-strings.
+
 2008-04-03  John W. Eaton  <jwe@octave.org>
 
 	* parse.y (make_constant): Also stash original text for strings.
 
 	* ov-fcn-handle.cc (octave_fcn_handle::subsref):
 	Don't call next_subsref here.
 
 2008-04-03  Jaroslav Hajek <highegg@gmail.com>
diff --git a/src/parse.y b/src/parse.y
--- a/src/parse.y
+++ b/src/parse.y
@@ -1729,22 +1729,29 @@ make_constant (int op, token *tok_val)
 	retval = new tree_constant (tmp, l, c);
 	retval->stash_original_text (tok_val->text_rep ());
       }
       break;
 
     case DQ_STRING:
     case SQ_STRING:
       {
+	std::string txt = tok_val->text ();
+
 	char delim = op == DQ_STRING ? '"' : '\'';
-	octave_value tmp (tok_val->text (), delim);
+
+	octave_value tmp (txt, delim);
 	retval = new tree_constant (tmp, l, c);
+
+	if (op == DQ_STRING)
+	  txt = undo_string_escapes (txt);
+
 	// FIXME -- maybe this should also be handled by
 	// tok_val->text_rep () for character strings?
-	retval->stash_original_text (delim + tok_val->text () + delim);
+	retval->stash_original_text (delim + txt + delim);
       }
       break;
 
     default:
       panic_impossible ();
       break;
     }
 
